contracts/utils/structs/EnumerableSet.sol

```diff
@@ -184,6 +184,13 @@ library EnumerableSet {
         return _at(set._inner, index);
     }
 
+    /**
+     * @dev Return the entier set in an array
+     */
+    function values(Bytes32Set storage set) internal view returns (bytes32[] memory) {
+        return set._inner._values;
+    }
```

Comment:
Note that returning a `memory` array implies reading the entire thing from storage and copying it - not sure if that was the original intent

Reply:
Also, shouldn't this be implemented for `Set` as `Set._values()`, and call that here?

Reply:
Returning a storage pointer will potentially encourage the users to write/modify it, which would break the structure. We really want to avoid this.

I'll consider adding a `Set._values()`. For it to make sense it should return a storage pointer otherwize specialized versions will not be able to do the slot trick for casting ... but as long as it is private, I'd be ok with that.